### PR TITLE
clean-up of test suite, merge testing dataframes

### PR DESCRIPTION
**Pull request by danielhuppmann**
# Description of PR

This PR merges two datasets (`test_df` and `meta_df`), which were almost identical and used interchangeably throughout the test suite.

### CODE REVIEW

**Comment by danielhuppmann** [comment]
@gidden & @znicholls, before diving into developing new features, I wanted to clean up the testing dataframes to improve the long-term sustainability of this project.

how to proceed:

0. review this PR, then make another PR for the next round of clean-ups
1. continue, then review all test updates in one PR
2. whatever, no need to review this (as it doesn't change any features or test logic)

**Comment by znicholls** [comment]
I’d do 0. In my opinion the tests are more important than the code and reviewing test changes is hard so should be done carefully. Of course there’s a balance and you have a better sense of the big picture so I’m happy to be overruled.

File: tests/test_cast_to_iamc.py [hunk 1]
```diff
@@ -4,43 +4,40 @@
...
 
-def test_cast_from_value_col(meta_df):
+
+def test_cast_from_value_col(test_df_year):
     df_with_value_cols = pd.DataFrame([
-        ['model_a', 'scen_a', 'World', 'EJ/y', TEST_DTS[0], 1, 0.5],
-        ['model_a', 'scen_a', 'World', 'EJ/y', TEST_DTS[1], 6., 3],
-        ['model_a', 'scen_b', 'World', 'EJ/y', TEST_DTS[0], 2, None],
-        ['model_a', 'scen_b', 'World', 'EJ/y', TEST_DTS[1], 7, None]
+        ['model_a', 'scen_a', 'World', 'EJ/y', 2005, 1, 0.5],
```
Comment (gidden) [inline tests/test_cast_to_iamc.py]: Is there a reason for preferring hardcoded values to referenced variables?

Comment (gidden) [reply tests/test_cast_to_iamc.py]: Same comments below as well

Comment (danielhuppmann) [reply tests/test_cast_to_iamc.py]: the previous implementation tested the same thing twice

0. create an `IamDataFrame` with year-columns
1. test it against the test-data with year column
2. test it again against the datetime-data converted to year columns, which is exactly the same object as 1.

So I simplified it by removing step 2, and then choose the shortest notation:
 `datetime(2005, 6, 17)` > `TEST_DTS[0]` > `2005`

**Pull request merged**